modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/IndexNotBuiltException.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.storage.index;
+
+import org.apache.ignite.internal.storage.StorageException;
+import org.apache.ignite.lang.ErrorGroups.Storage;
+
+/** Exception occurred while reading from an index that has not yet been built. */
```

Comment:
```suggestion
/** Exception occurring while reading from an index that has not yet been built. */
```

Reply:
Should it be mentioned in the javadoc that this might be a result of data loss (like majority loss), so that the Catalog thinks the index built was finished, but the partition does not have the index fully built?

Reply:
Fixed it. 
I don’t think we need to talk about this here; an error can also occur if the API is used incorrectly.